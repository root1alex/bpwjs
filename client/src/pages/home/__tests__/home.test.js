import configureMockStore from 'redux-mock-store';

import HomePage, {Home} from '../index';

const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: new Date(2017, 1, 1, 1, 1, 1, 1),
};

test('# Home page wrapper', () => {
  const store = mockStore({auth: {user}, questions: {questions: []}});
  const wrapper = shallow(<HomePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Home page', () => {
  const questions = [];
  const fetchQuestions = () => {
    questions.push({
      id: '0',
      owner: user,
      text: 'Question text',
      answers: [{answer: 'Text answer'}],
    });
  };

  const component = (
    <Home
      user={user}
      questions={questions}
      fetchQuestions={fetchQuestions}
      doAnswer={() => {}}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
