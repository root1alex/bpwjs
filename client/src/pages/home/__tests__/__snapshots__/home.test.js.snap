// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Home page 1`] = `
<div>
  <Component
    current="/"
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": 2017-01-31T21:01:01.001Z,
      }
    }
  />
  <div>
    <Connect(Question)
      onAnswer={[Function]}
      question={
        Object {
          "answers": Array [
            Object {
              "answer": "Text answer",
            },
          ],
          "id": "0",
          "owner": Object {
            "id": "0",
            "login": "test",
            "registrationDate": 2017-01-31T21:01:01.001Z,
          },
          "text": "Question text",
        }
      }
    />
  </div>
</div>
`;

exports[`# Home page wrapper 1`] = `
<Home
  doAnswer={[Function]}
  fetchQuestions={[Function]}
  questions={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": 2017-01-31T21:01:01.001Z,
    }
  }
/>
`;
